### ISSUE TITLE
SQL installer not working properly or not giving out log files?

### ISSUE THREAD

**Issue by riahc3** [ryancbutler/Citrix-VAD-LAB#3]
Hello

Based on your Ansible playbook, I made this one, where the part of installing SQL Server 2017 is 98% the same:

---
- hosts: "HOSTWITHSQL"

  vars:
    sql_svc_account: ad.mydomain.lan\svcsql
    sql_svc_password: svcsql
    sql_admins: ad.mydomain.lan\sqladmins
    sql_instance: sqlcitrix
    
  tasks:
  - name: "Create SQL Path"
    win_file:
     path: C:\Temp\SQL\Files
     state: directory

  - name: "Copy SQL Files To Host"
    win_copy:
     src: "/ansiblefiles/isos/SqlSetup_x64_ENU"
     dest: C:\Temp\SQL\Files
     backup: yes
    register: dest
    
  - name: Install SQL Components
    win_package:
     path: "C:\\Temp\\SQL\\Files\\SqlSetup_x64_ENU\\setup.exe"
     arguments: /q /ACTION=Install /FEATURES=SQL /INSTANCENAME="{{ sql_instance }}" /SQLSVCACCOUNT="{{ sql_svc_account }}" /SQLSVCPASSWORD="{{ sql_svc_password }}" /SQLSYSADMINACCOUNTS="{{ sql_admins }}" /AGTSVCACCOUNT="NT AUTHORITY\Network Service" /SQLSVCINSTANTFILEINIT="True" /IACCEPTSQLSERVERLICENSETERMS /UpdateEnabled=0
     state: present
     creates_path: C:\Program Files\Microsoft SQL Server
    register: sql_install

  - name: Reboot after SQL
    win_reboot:
    when: sql_install.changed

  - name: Install SQL Management
    win_chocolatey:
     name: sql-server-management-studio
     state: present
    
  - name: "Delete files folder"
    win_file:
     path: C:\Temp\SQL\Files
     state: absent

The weird part is that the setup.exe opens for < 1 sec in Task Manager and then closes. Ansible says it installed correctly.

From what Ive been reading, the setup SHOULD dump SOME kind of log file: https://docs.microsoft.com/en-us/sql/database-engine/install-windows/view-and-read-sql-server-setup-log-files?view=sql-server-2017 but Ive searched and I didnt find anything. 

Is it possible to change the arguments to include a log file output of the setup?

Thank you

**Comment by ryancbutler** [comment]
This probably means the setup isn't getting very far.  I would check the logs in C:\Program Files\Microsoft SQL Server and also try running the command from the server without Ansible to see what it's doing.  If you are creating a new instance the site creation script will also need to be modified.

**Comment by riahc3** [comment]
> This probably means the setup isn't getting very far. I would check the logs in C:\Program Files\Microsoft SQL Server and also try running the command from the server without Ansible to see what it's doing. 

Yup, I tried both these things. Sadly, without ansible I can get it to work without issues. Im starting to think its a quote issue....

> If you are creating a new instance the site creation script will also need to be modified.

Yup, I have this written in my notes. Thank you for the reminder nonetheless!

Well, this is weird: If a Microsoft SQL Server folder exists, the installation fails (plus the Setup Bootstrap isnt even in there). If I rename it xMicrosoft SQL Server, it creates a new Microsoft SQL Server folder (as stated in the playbook) and then the installation works fine. Really odd.

Is there a reason you dont put the server to start Automatic? I went ahead and added that.
